### PR TITLE
Integrate Event list into User model and implement persistent storage

### LINKED ISSUE #90: Integrate Events into User model
have the events list be stored in the User object.

### LINKED ISSUE #120: Implement persistent storage for Events in User

### PR DESCRIPTION
**Pull request by igezt**
Integrate Events into the User Model and implemented persistent storage as well

* Closes #90 
* Closes #120

### CODE REVIEW

File: src/main/java/seedu/address/model/event/fields/Recurrence.java [hunk 1]
```diff
@@ -53,9 +53,10 @@ public Recurrence(String interval) {
...
      * Returns if a given string is a valid Recurrence.
      */
-    public static boolean isValidInterval(String trimmedInterval) {
+    public static boolean isValidRecurrence(String trimmedInterval) {
         String lowerCaseInterval = trimmedInterval.toLowerCase(Locale.ROOT);
-        return Objects.equals(lowerCaseInterval, DAILY_CASE)
+        return Objects.equals(lowerCaseInterval, NONE_CASE)
+                || Objects.equals(lowerCaseInterval, DAILY_CASE)
                 || Objects.equals(lowerCaseInterval, WEEKLY_CASE)
                 || Objects.equals(lowerCaseInterval, MONTHLY_CASE) || Objects.equals(lowerCaseInterval, YEARLY_CASE);
```
Comment (hansstanley) [inline src/main/java/seedu/address/model/event/fields/Recurrence.java:61]: Could be effectively replaced with:

```suggestion
        return List.of(NONE_CASE, DAILY_CASE, WEEKLY_CASE, MONTHLY_CASE, YEARLY_CASE)
                .contains(lowerCaseInterval);
```

**Review by hansstanley — APPROVED** [review]
Minor changes but LGTM

File: src/main/java/seedu/address/model/event/fields/Recurrence.java (hunk 1 above)
Comment (igezt) [reply src/main/java/seedu/address/model/event/fields/Recurrence.java:61]: stonks.

**Comment by hansstanley** [comment]
* Closes #35 
* Closes #44

**Pull request merged**